Sources/StreamChatUI/ChatThread/ChatThreadVC.swift

```diff
@@ -258,6 +258,22 @@ open class ChatThreadVC: _ViewController,
         messageComposerVC.dismissSuggestions()
     }
 
+    open func chatMessageListVC(
+        _ vc: ChatMessageListVC,
+        decorationViewForMessage message: ChatMessage,
+        decorationType: ChatMessageDecorationType,
+        at indexPath: IndexPath
+    ) -> ChatMessageDecorationView? {
+        switch decorationType {
+        case .header where vc.shouldShowDateSeparator(forMessage: message, at: indexPath):
+            let dateSeparatorView = components.messageListDateSeparatorView.init()
+            dateSeparatorView.content = vc.dateSeparatorFormatter.format(message.createdAt)
+            return dateSeparatorView
+        default:
+            return nil
+        }
+    }
```

Comment:
Another problem with this, is that it means we need to duplicate 3 times the default implementation right now 🤔 It might be worth it to export this default implementation into a component, where we just call it in the 3 different implementations. Then for specific things related to the Channel or Thread, it will be handled only there 🤔 Just thinking out loud here. But ideally, for every non-channel/thread related separators, the Channel, Thread or a Custom implementation of the Message List, would get them for fre

Reply:
Any implementation of `ChatMessageListVCDelegate` the users have they will need to implement the 2 new methods. So, what we can do is to provide a method on an extension of `ChatMessageListVCDelegate` that can be used to create the DateSeparator. 
I don't see a need to default implement the 2 new methods. WDYT?

Reply:
We need to provide a default implementation, unfortunately. Otherwise this won't compile for customers, which is a breaking change, and would require a major bump. Protocols in SDKs can be problematic, especially if the API is not final. 

Another problem here, is that customers already had the date separators out of the box with the `ChatMessageListVC`, and with this change, they don't. So we really need to provide this implementation out of the box.